### PR TITLE
plugins/dnf: implement validate_certs option

### PR DESCRIPTION
**Pull request by philfry**
##### SUMMARY

This PR implements the `validate_certs` option of the `dnf` module. As the option is mentioned in the documentation, this is not a feature PR but a bugfix PR.

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

dnf

##### ADDITIONAL INFORMATION

See #76355

### CODE REVIEW

**Comment by mkrizek** [comment]
We have discussed this and thought it would be better to add a new option (`sslverify`) instead of reusing the existing one that serves a different purpose.

This would also need a changelog. See https://docs.ansible.com/ansible/devel/community/development_process.html#changelogs-how-to

Ideally this would be added to the `yum` module as well to keep the parity of functionality.

**Comment by philfry** [comment]
Thanks. I'll update the PR once tested.

**Comment by philfry** [comment]
Tested so far against a CentOS 7 with yum and against an OracleLinux 8 with dnf using a broken certificate (not valid for this servername). Works for me.

File: lib/ansible/modules/dnf.py [hunk 1]
```diff
@@ -172,6 +172,13 @@
     type: bool
     default: "yes"
     version_added: "2.7"
+  sslverify:
+    description:
+      - Disables SSL validation of the repository server for this transaction.
+      - This should be set to C(no) if one of the configured repositories is using an untrusted or self-signed certificate.
+    type: bool
+    default: "yes"
+    version_added: "2.12"
```
Comment (mkrizek) [inline lib/ansible/modules/dnf.py]: This would have to be `2.13` as `2.12` is released already and we do not backport features. Same below.

Comment (philfry) [reply lib/ansible/modules/dnf.py]: thanks, I fixed it.

**Comment by mkrizek** [comment]
There are failures in CI for RHEL 7 and CentOS 7 which suggest there is an issue with the implementation for the `yum` module:
* https://dev.azure.com/ansible/ansible/_build/results?buildId=30874&view=logs&j=f23d423e-3f98-549c-dd57-43996eeae720&t=a23c0f81-683a-569c-1226-9f1dc61ddefb
* https://dev.azure.com/ansible/ansible/_build/results?buildId=30874&view=logs&j=c055dccd-56e7-5942-cfff-a419afaadc39&t=3158b048-5702-58c0-2e6b-4829f39cd4f9

**Comment by philfry** [comment]
> There are failures in CI for RHEL 7 and CentOS 7 which suggest there is an issue with the implementation for the `yum` module:

looks like it's fixed now.

**Comment by philfry** [comment]
hold it, something's not right yet.

**Review by mkrizek — APPROVED** [review]

**Pull request merged**